File: FPG.Core/Factories/SaltingAlgorithmFactory.cs
using System;

using FPG.Algorithms;
using FPG.Interfaces;

namespace FPG.Factories
{
    public static class SaltingAlgorithmFactory
    {
        public static ISaltingAlgorithm Create(String name)
        {
            switch (name.ToLower())
            {
                case "additive":
                    return new AdditiveSaltingAlgorithm();
                default:
                    throw new ArgumentException("Unknown algorithm name");
            };
        }
    }
}

File: lib/core/factories/SaltingAlgorithmFactory.dart
import 'package:fpg/core/algorithms/salting/AdditiveSaltingAlgorithm.dart';
import 'package:fpg/core/interfaces/ISaltingAlgorithm.dart';

class SaltingAlgorithmFactory {
  static ISaltingAlgorithm create(String name) {
    switch (name.toLowerCase()) {
      case "additive":
        return AdditiveSaltingAlgorithm();
      default:
        throw FormatException("Unknown algorithm name");
    }
  }
}
